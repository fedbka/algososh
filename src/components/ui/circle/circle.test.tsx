import { expect, describe, test } from "vitest";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

import { render } from "@testing-library/react";

const TEST_DATA = {
  letter: 'CTD',
  index: 1,
  defaultState: ElementStates.Default,
  changingState: ElementStates.Changing,
  modifiedState: ElementStates.Modified,
  reactElement: <Circle letter="'CTD'" isSmall />,
}

describe("Тестирование компонента 'Circle'", () => {

  test("Компонент 'Circle' без буквы формируется без ошибок", () => {
    const { container } = render(<Circle />);
    expect(container).toMatchSnapshot();
  });

  test("Компонент 'Circle' с буквой формируется без ошибок", () => {
    const { container } = render(<Circle letter={TEST_DATA.letter} />);
    expect(container).toMatchSnapshot();
  });

  test("Компонент 'Circle' с буквой в head формируется без ошибок", () => {
    const { container } = render(<Circle head={TEST_DATA.letter} />);
    expect(container).toMatchSnapshot();
  });

  test("Компонент 'Circle' с React-элементом в head формируется без ошибок", () => {
    const { container } = render(<Circle head={TEST_DATA.reactElement} />);
    expect(container).toMatchSnapshot();
  });

  test("Компонент 'Circle' с буквой в tail формируется без ошибок", () => {
    const { container } = render(<Circle tail={TEST_DATA.letter} />);
    expect(container).toMatchSnapshot();
  });

  test("Компонент 'Circle' с React-элементом в tail формируется без ошибок", () => {
    const { container } = render(<Circle tail={TEST_DATA.reactElement} />);
    expect(container).toMatchSnapshot();
  });

  test("Компонент 'Circle' с индексом формируется без ошибок", () => {
    const { container } = render(<Circle index={TEST_DATA.index} />);
    expect(container).toMatchSnapshot();
  });

  test("Компонент 'Circle' с указанным свойством 'isSmall' формируется без ошибок", () => {
    const { container } = render(<Circle isSmall />);
    expect(container).toMatchSnapshot();
  });
  
  test("Компонент 'Circle' с указанным состояние 'ElementStates.Default' формируется без ошибок", () => {
    const { container } = render(<Circle state={TEST_DATA.defaultState} />);
    expect(container).toMatchSnapshot();
  });
  
  test("Компонент 'Circle' с указанным состояние 'ElementStates.Changing' формируется без ошибок", () => {
    const { container } = render(<Circle state={TEST_DATA.changingState} />);
    expect(container).toMatchSnapshot();
  }); 
  
  test("Компонент 'Circle' с указанным состояние 'ElementStates.Modified' формируется без ошибок", () => {
    const { container } = render(<Circle state={TEST_DATA.modifiedState} />);
    expect(container).toMatchSnapshot();
  });  

});