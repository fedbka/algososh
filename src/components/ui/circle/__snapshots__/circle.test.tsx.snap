// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' без буквы формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63   _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    />
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с React-элементом в head формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    >
      <div
        class="_content_c89a63 "
      >
        <div
          class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
        />
        <div
          class="_circle_c89a63  _small_c89a63 _default_c89a63"
        >
          <p
            class="text text_type_circle text_color_input _letter_c89a63"
          >
            'CTD'
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
        />
        <div
          class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
        />
      </div>
    </div>
    <div
      class="_circle_c89a63   _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    />
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с React-элементом в tail формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63   _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    >
      <div
        class="_content_c89a63 "
      >
        <div
          class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
        />
        <div
          class="_circle_c89a63  _small_c89a63 _default_c89a63"
        >
          <p
            class="text text_type_circle text_color_input _letter_c89a63"
          >
            'CTD'
          </p>
        </div>
        <p
          class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
        />
        <div
          class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с буквой в head формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _string_c89a63"
    >
      CTD
    </div>
    <div
      class="_circle_c89a63   _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    />
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с буквой в tail формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63   _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _string_c89a63"
    >
      CTD
    </div>
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с буквой формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63   _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      >
        CTD
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    />
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с индексом формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63   _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    >
      1
    </p>
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail60_c89a63 _element_c89a63"
    />
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с указанным свойством 'isSmall' формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63  _small_c89a63 _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    />
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с указанным состояние 'ElementStates.Changing' формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63   _changing_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    />
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с указанным состояние 'ElementStates.Default' формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63   _default_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    />
  </div>
</div>
`;

exports[`Тестирование компонента 'Circle' > Компонент 'Circle' с указанным состояние 'ElementStates.Modified' формируется без ошибок 1`] = `
<div>
  <div
    class="_content_c89a63 "
  >
    <div
      class="text text_type_input text_color_input mb-4 _absolute_c89a63 _head_c89a63 _element_c89a63"
    />
    <div
      class="_circle_c89a63   _modified_c89a63"
    >
      <p
        class="text text_type_circle text_color_input _letter_c89a63"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _index_c89a63"
    />
    <div
      class="text text_type_input text_color_input mt-4 _absolute_c89a63 _tail30_c89a63 _element_c89a63"
    />
  </div>
</div>
`;
